doc/TROUBLESHOOT.md

```diff
@@ -4,24 +4,51 @@ The text below is under construction.
 
 TODO: separate information for tests and applications.
 
-# Memory Leak Troubleshooting
+# Troubleshoot Memory Leaks
 
-This page describes how to troubleshoot memory leaks. See leak detection guidance [here](DETECT.md).
+This page describes how to troubleshoot memory leaks. See other information on memory leaks [here](../README.md).
 
-## Collect callstack
+To understand root cause of a memroy leak, gather additional information.
 
-Stacktrace for the object's lifecycle events may help to catch out
-the leak's root cause. The lifecycle event will be creation
-for not-disposed leaks, and disposal for non-GCed leaks.
+- **not-disposed**:
+
+    - **Allocation call-stack** helps to detect
+        owner of the object, that is responsible for the object disposal.
+
+- **not-GCed or GCed-late**:
+
+    - **Allocation and disposal call-stacks**: help to understand lifecicle of the
```

Comment:
r/help/helps
sp: lifecycle
'object, which may reveal where the object is being held from garbage collection'